src/Documentation/tutorials_and_samples/tutorial_1_helloworld.md

```diff
@@ -0,0 +1,139 @@
+
+This process ties into the Hello World sample application available [here](https://github.com/dotnet/orleans/tree/master/Samples/2.0/HelloWorld).
+
+The main concepts of Orleans involve a silo, a client, and one or more grains.
+Creating an Orleans app involves configuring the silo, configuring the client, and writing the grains.
+
+## Configuring the silo
+
+Silos are configured programmatically via `SiloHostBuilder` and a number of supplemental option classes.
+A list of all of the options can be found [here.](http://dotnet.github.io/orleans/Documentation/clusters_and_clients/configuration_guide/list_of_options_classes.html)
+
+```csharp
+[...]
+        private static async Task<ISiloHost> StartSilo()
+        {
+            // define the cluster configuration
+            var builder = new SiloHostBuilder()
+                .UseLocalhostClustering()
+                .Configure<ClusterOptions>(options =>
+                {
+                    options.ClusterId = "dev";
+                    options.ServiceId = "HelloWorldApp";
+                })
+                .Configure<EndpointOptions>(options => options.AdvertisedIPAddress = IPAddress.Loopback)
+                .ConfigureApplicationParts(parts => parts.AddApplicationPart(typeof(HelloGrain).Assembly).WithReferences())
+                .ConfigureLogging(logging => logging.AddConsole());
+
+            var host = builder.Build();
+            await host.StartAsync();
+            return host;
+        }
+```
+
+| Option | Used for |
+|-------------|----------|
+| `.UseLocalhostClustering()` | Declaring that we are using a single local silo |
+| `ClusterOptions` | ClusterId is the name for the Orleans cluster must be the same for silo and client so they can talk to each other. ServiceId is the ID used for the application and it must not change across deployments|
+| `EndpointOptions` | This tells the silo where to listen. For this example, we are using a `loopback`. |
+| `ConfigureApplicationParts` | Adds the assembly with grain classes to the application setup. |
+
+After loading the configurations, the SiloHost is built and then started asynchronously.
+
+## Configuring the client
+
+Similar to the silo, the client is configured via `ClientBuilder` and a similar collection of option classes.
+
+```csharp
+        private static async Task<IClusterClient> StartClientWithRetries()
+        {
+            attempt = 0;
+            IClusterClient client;
+            client = new ClientBuilder()
+                .UseLocalhostClustering()
+                .Configure<ClusterOptions>(options =>
+                {
+                    options.ClusterId = "dev";
+                    options.ServiceId = "HelloWorldApp";
+                })
+                .ConfigureLogging(logging => logging.AddConsole())
+                .Build();
+
+            await client.Connect(RetryFilter);
+            Console.WriteLine("Client successfully connect to silo host");
+            return client;
+        }
```

Comment:
also if we could paste the client program here, that would be helpful for people to figure out the flow of this sample, which you described in "How parts work together" 

```
private static async Task DoClientWork(IClusterClient client)
        {
            // example of calling grains from the initialized client
            var friend = client.GetGrain<IHello>(0);
            var response = await friend.SayHello("Good morning, my friend!");
            Console.WriteLine("\n\n{0}\n\n", response);
        }
```

ofc please explain what this block of code is doing.

Reply:
I put the code closer to the section that explains the process. I think it should make a lot more sense now.